app/abci/handlers.go

```diff
@@ -322,13 +321,8 @@ func (h *Handlers) PreBlocker() sdk.PreBlocker {
 			if err != nil {
 				return nil, err
 			}
-			batchSigs := batchingtypes.BatchSignatures{
-				BatchNumber:   batchNum,
-				ValidatorAddr: validator.OperatorAddress,
-				Signatures:    vote.VoteExtension,
-			}
 
-			err = h.batchingKeeper.SetBatchSignatures(ctx, batchSigs)
+			err = h.batchingKeeper.SetBatchSigSecp256k1(ctx, batchNum, validator.OperatorAddress, vote.VoteExtension)
```

Comment:
Great catch.. When a validator unbonds completely and is removed from the validator set, this does return an error as expected. The error doesn't halt the chain as it is captured by the PreBlock's error handler. Interestingly, the other validators' batch signatures nonetheless get stored correctly..

Reply:
I encountered an issue as I was implementing a separate store for batch signatures because, when storing them, we don't have access to the corresponding scheme's public keys. I am trying to solve this by creating batch signature entries for the next batch when we construct a batch and then later populating them with signatures. I think I will need more time on this though. This might also be easier when tackled alongside the new bootstrapping plan. Pushed a WIP commit to the branch `hy/batch-sig-refactor`.

Reply:
I suppose we should just store the public keys in validator entry store and signatures separately in the batch signature store.. My bad.. Will make that change in this PR.